src/main/java/core/basesyntax/service/db/StorageServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package core.basesyntax.service.db;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class StorageServiceImpl implements StorageService {
+    private final Map<String, Integer> storage = new HashMap<>();
+
+    @Override
+    public void updateBalance(String fruit, int amount) {
+        storage.put(fruit, amount);
+    }
+
+    @Override
+    public void addSupply(String fruit, int amount) {
+        storage.put(fruit, storage.getOrDefault(fruit, 0) + amount);
+    }
+
+    @Override
+    public void purchaseFruit(String fruit, int amount) {
+        if (storage.containsKey(fruit)) {
+            int currentQuantity = storage.get(fruit);
+            if (currentQuantity >= amount) {
+                storage.put(fruit, currentQuantity - amount);
+            } else {
+                throw new RuntimeException("Not enough fruit in stock to complete the purchase.");
+            }
+        } else {
+            throw new RuntimeException("No such fruit in stock.");
+        }
```

Comment:
This is a code quality issue. You should check the fruit balance before saving it in the Storage - it should be positive. Throw `RuntimeException` in case the balance is negative.